File: api/web/models/chat_log.py
"""APIs for ChatLog"""
from typing import Dict, Optional, Any
from modules.types import PMObject


class ChatLogInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    role: str
    name: Optional[str] = None
    content: Optional[str] = None
    tool_calls: Optional[Dict[str, Any]] = None

    token_usage: Optional[int] = None

    latency: Optional[float] = None
    cost: Optional[float] = None
    chat_log_metadata: Optional[Dict[str, Any]] = None

    session_uuid: str


class ChatLogViewInstance(PMObject):
    assistant_log_id: int
    project_uuid: str
    chat_model_uuid: str
    chat_model_name: str

    chat_model_version_uuid: str
    chat_model_version: int

    session_uuid: str

    created_at: str
    user_input: str
    assistant_output: Optional[str] = None
    tool_calls: Optional[Dict[str, Any]] = None

    latency: Optional[float] = None
    cost: Optional[float] = None
    token_usage: Optional[Dict[str, Any]] = None

    run_from_deployment: bool


class ChatLogCountInstance(PMObject):
    project_uuid: str
    count: int

File: src/types/ChatLog.ts
// Interfaces for ChatLog retrieval processes

/**
 * Interface for request to read ChatModelVersion's ChatLogSessions.
 */
export interface ReadSessionChatLogsRequest {
  chat_session_uuid: string;
  page: number;
  rows_per_page: number;
}

/**
 * Interface for request to read Project's ChatLogSessions.
 */
export interface ReadProjectChatLogsRequest {
  project_uuid: string;
  page: number;
  rows_per_page: number;
}

/**
 * Interface for request to read Project's ChatLogs count.
 */
export interface ReadProjectChatLogsCountRequest {
  project_uuid: string;
}

/**
 * Interface for response of ChatLogs count.
 */
export interface ReadProjectChatLogsCountResponse {
  project_uuid: string;
  count: number;
}

/**
 * General interface for ChatLog.
 */
export interface ChatLog {
  id?: number;
  created_at: string;
  uuid?: string;
  role: string;
  name?: string;
  content: string;
  tool_calls?: Record<string, any>;
  token_usage?: number;
  latency?: number;
  cost?: number;
  chat_log_metadata?: Record<string, any>;
  session_uuid?: string;
}

/**
 * General interface for ChatLogView.
 */
export interface ChatLogView {
  assistant_log_id: number;
  project_uuid: string;
  chat_model_uuid: string;
  chat_model_name: string;
  chat_model_version_uuid: string;
  chat_model_version: number;
  session_uuid: string;
  created_at: string;
  user_input: string;
  assistant_output: string;
  tool_calls: Record<string, any>;
  latency: number;
  cost: number;
  token_usage: Record<string, any>;
  run_from_deployment: boolean;
}
